src/App.tsx

```diff
@@ -1,10 +1,47 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+
+type FilterType = 'all' | 'active' | 'completed';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState<FilterType>('all');
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setError('');
+    setIsLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => setError('Unable to load todos'))
+      .finally(() => {
+        setIsLoading(false);
+        setTimeout(() => setError(null), 3000);
+      });
+  }, []);
+
+  const filteredTodos = useMemo(
```

Comment:
You can split code of this component. 
Please try use clear function, and split code.